### PR TITLE
Veraison demo

### PR DESCRIPTION
**Pull request by Havner**
Veraison demo

- moving /examples/veraison from islet-ra repo to islet
- adapting code/scripts for the veraison demo
- updating veraison demo readme
- making sure everything builds and works

### CODE REVIEW

**Comment by TomaszSwierczek** [comment]
@bokdeuk-jeong , still WIP but we're finalizing it. Should be done tomorrow & if nothing blows up, tested too. Again, sorry for mess with changing the plan, this is aligned to : https://github.com/Samsung/islet/pull/236#issuecomment-1827548375 like you asked.

**Comment by zpzigi754** [comment]
FYI, rebasing to the main branch will pass the failed workflow (rule-checker).

**Comment by Havner** [comment]
Fixed, verified, ready to merge (requires https://github.com/Samsung/islet/pull/248)

**Review by zos — DISMISSED** [review]

File: examples/veraison/README.md [hunk 1]
```diff
@@ -1,6 +1,325 @@
...
+
+The following repos will be used:
+
+* Islet: https://github.com/Samsung/islet that provides:
+
+	* the whole SW/FW stack and scripts for running the emulated environment under the FVP
+	* Islet HES https://github.com/Samsung/islet/tree/main/hes
+	* kvmtool-rim-measurer from https://github.com/Samsung/islet/tree/main/third-party/
+
+* Islet Remote Attestation: https://github.com/Samsung/islet-ra that provides:
```
Comment (bokdeuk-jeong) [inline examples/veraison/README.md:23]: islet-ra will appear as a submodule of islet. Do we need to mention this URL?
Metioning each tool in the perspective directoy appearing in islet looks better.

File: examples/veraison/README.md [hunk 2]
```diff
@@ -1,6 +1,325 @@
...
+
+    * please use 5a48655b3a9c3960667ef14df7860186238b6bcd commit. Newer versions
+      might not work at the moment
+
+# Preparation
+
+The 3 beforementioned repositories should be checked out on the same level so it
+should look like following:
+
+    CCA/islet
```
Comment (bokdeuk-jeong) [inline examples/veraison/README.md:40]: Cloning islet repo does dont present CCA directory. This makes readers confused.

File: examples/veraison/README.md [hunk 3]
```diff
@@ -1,6 +1,325 @@
...
+	CCA/islet-ra
+	CCA/services   (this is veraison repository)
+
+Install ubuntu's aarch64 compiler, it's needed by some tools:
+
+    # apt install gcc-aarch64-linux-gnu
```
[the remark this answers is not in the document]
Comment (bitboom) [reply examples/veraison/README.md:46]: We uses `aarch64-gcc v11.3` located in `islet/assets` to build our components. 
So it doesn't located in our [init scripts](https://github.com/Samsung/islet/blob/main/scripts/deps/pkgs.sh) now.

But I think it's better to  add it to our [init scripts](https://github.com/Samsung/islet/blob/main/scripts/deps/pkgs.sh) to build 3rd-parties components as you suggested,

File: examples/veraison/README.md [hunk 4]
```diff
@@ -1,6 +1,325 @@
...
+
+* Islet: https://github.com/Samsung/islet that provides:
+
+	* the whole SW/FW stack and scripts for running the emulated environment under the FVP
+	* Islet HES https://github.com/Samsung/islet/tree/main/hes
+	* kvmtool-rim-measurer from https://github.com/Samsung/islet/tree/main/third-party/
+
+* Islet Remote Attestation: https://github.com/Samsung/islet-ra that provides:
+
+	* rocli: https://github.com/Samsung/islet-ra/tree/main/tools/rocli
```
[the remark this answers is not in the document]
Comment (Havner) [reply examples/veraison/README.md:25]: Will do.

File: examples/veraison/README.md (hunk 2 above)
Comment (Havner) [reply examples/veraison/README.md:40]: The idea for this directory is explained at the very beginning of the document (Introduction, line 13). I wanted it to serve as a common root for all the required repositories and in my opinion having that and seeing all the paths coming from common root actually reduces confusion.

File: examples/veraison/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Havner) [reply examples/veraison/README.md:46]: The thing is that I tried to compile the tools with the aarch64-gcc that is included in islet but it failed to do so. The compiler provided by Ubuntu works. Regarding adding it to init.sh I'm all for it. Didn't do that as it's not required by islet itself, just by the demo. But if we all agree on this I'll do that.

File: examples/veraison/README.md [hunk 5]
```diff
@@ -1,6 +1,325 @@
...
+	  A library implementing token and RSI related functionalities (fetching, parsing).
+
+* veraison: https://github.com/veraison/services
+
+    * Please use 5a48655b3a9c3960667ef14df7860186238b6bcd commit. Newer versions
+      might not work at the moment.
+
+# Preparation
+
+The 3 beforementioned repositories should be checked out on the same level so it
```
Comment (bitboom) [inline examples/veraison/README.md:41]: minor spacing

```suggestion
The 3 before mentioned repositories should be checked out on the same level so it
```

**Review by bitboom — APPROVED** [review]
Nice attestation demo 👍

File: examples/veraison/realm-application/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+[workspace]
+
+[package]
+name = "realm-application"
+version = "0.1.0"
+edition = "2021"
+authors = [ "Lukasz Pawelczyk" ]
```
[the remark this answers is not in the document]
Comment (Havner) [reply examples/veraison/realm-application/Cargo.toml]: Done

File: examples/veraison/provisioning/README.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
+# Compile veraison from the CCA demo tutorial
+
+[Link to CCA Demo staeps from veraison docs](https://github.com/veraison/docs/blob/main/demo/cca/manual-end-to-end.md)
```
[the remark this answers is not in the document]
Comment (Havner) [reply examples/veraison/provisioning/README.md]: Done

**Review by bokdeuk-jeong — APPROVED** [review]

**Pull request merged**